resource-directory/service/resourceShadow.go

```diff
@@ -166,7 +177,8 @@ func (rd *ResourceShadow) RetrieveDevicesMetadata(req *pb.RetrieveDevicesMetadat
 	}
 
 	if len(devicesMetadata) == 0 {
-		return status.Errorf(codes.NotFound, "not found")
+		log.Debug("ResourceShadow.GetDevicesMetadata.filterMetadataByUserFilters returns empty devicesMetadata")
```

Comment:
same here. + returned empty device metadata.

Reply:
just to know which filter returns empty data